static/app/views/settings/components/forms/selectAsyncField.tsx

```diff
@@ -0,0 +1,64 @@
+import * as React from 'react';
+
+import SelectAsyncControl, {
+  Props as SelectAsyncControlProps,
+} from 'app/components/forms/selectAsyncControl';
+import InputField from 'app/views/settings/components/forms/inputField';
+
+//projects can be passed as a direct prop as well
+type Props = InputField['props'];
+
+type SelectAsyncFieldProps = SelectAsyncControlProps & Props;
+
+class SelectAsyncField extends React.Component<SelectAsyncFieldProps> {
```

Comment:
could you maybe give more information on why we need to create a new component for this? Is this component only going to be used by us or was it built to replace other components going forward?

I just would have thought that getting a list of teams or users is a common thing we do in sentry so kind of surprised there was no existing components that we could reuse for our case

Reply:
@MeredithAnya This UI pattern is not common in Sentry. Generally, we fetch all the results and then render that in the SelectField as options. `SelectAsyncField` reuses the logic in `SelectAsyncControl` and lets us use it as a fieldFromConfig.